Detector/DetDriftChamber/src/driftchamber/DriftChamber.cpp

```diff
@@ -130,6 +130,39 @@ static dd4hep::Ref_t create_detector(dd4hep::Detector& theDetector,
 
         dd4hep::Tube layer_solid(rmin,rmax,chamber_length*0.5);
         dd4hep::Volume layer_vol(layer_name,layer_solid,det_mat);
+        layer_vol.setAttributes(theDetector,x_det.regionStr(),x_det.limitsStr(),x_det.visStr());
+
+        // - wire vol
+//        if(layer_id==0) {
+        for(int icell=0; icell< numWire; icell++) {
+            double wire_phi = (icell+0.5)*layer_Phi + offset;
+            for(xml_coll_t c(x_det,_U(module)); c; ++c) {
+                xml_comp_t x_module = c;
+                if(x_module.id()==0) {
+                  double module_rmin = x_module.rmin();
+                  double module_rmax = x_module.rmax();
+                  std::string module_name = x_module.nameStr();
+                  dd4hep::Tube module_solid(module_rmin,module_rmax,chamber_length*0.5);
+                  std::string Module_name = layer_name + module_name;
+                  dd4hep::Volume module_vol(Module_name,module_solid,det_mat);
```

Comment:
I think the logical volume of wires should be same. The only differences are the physical volumes. 
Can you try to construct the logical volume of the wire only once?

Reply:
Because it also contains plating, the logical volume is different.

Reply:
No, I mean for the signal wires. I think you don't construct the field wires yet.
Tao

Reply:
Yes, I only put the signal wire, because the distribution of the field wire has not been given yet.

Reply:
So could you optimize this code? 
Tao

Reply:
Yes, I can.